### DISCUSSION TITLE
What exactly happens when tgcf is run?

### DISCUSSION THREAD

**Discussion by User087** [aahnik/tgcf#234] [General]
Excuse me if this is a stupid question, but as a matter of curiosity, what exactly happens when e.g. the command 'tgcf --loud' is run? What files does it actually execute and in what order? And where is the action and order for executing files even defined? Looking at the output in the terminal, it would seem that the function `main` defined in tgcf/cli.py (or the line `@app.command()` or something) is first executed, then somewhow tgcf/config.py is executed, then live.py (if the mode is 'live'); is that correct? If so, where is this action of the command 'tgcf' defined? Does it get defined upon installing the dependencies using poetry? What is the exact definition given to 'tgcf'?

**Comment by aahnik** [comment]
Working on some major updates to tgcf.

After that will make multiple YouTube videos:

- tutorial to use and deploy tgcf
- the programming and design of tgcf, how it works, it's history, and future

**Comment by aahnik** [comment]
I love your curiosity. I am also in similar position when I see many projects.

Come, let's have a chat on telegram. Let's connect!

https://aahnik.dev
